tray.c

```diff
@@ -0,0 +1,58 @@
+// gcc `pkg-config appindicator3-0.1 --cflags` tray.c -o build/tray `pkg-config
+// appindicator3-0.1 --libs`
+#include "tray.h"
+
+#include <gtk/gtk.h>
+#include <libappindicator/app-indicator.h>
+
+#define ICON_NAME_EN "indicator-keyboard-En"
+#define ICON_NAME_VI "indicator-keyboard-Vi"
+
+static void quit_activated(GSimpleAction *simple, GVariant *parameter,
+			   gpointer user_data)
+{
+	struct daklakwl_state *state;
+	state = user_data;
+	int efd = wl_display_get_fd(state->wl_display);
+	state->running = false;
+	char *quit = "daklak.quit";
+	write(efd, &quit, strlen(quit));
```

Comment:
hi @tadeokondrak, there is an event polling in `daklakwl_state_run` which block the program exit so I make a hack here. Could you give me better advise?